Realm/Realm/Handles/ObjectHandle.cs

```diff
@@ -191,9 +191,18 @@ public void SetValue(string propertyName, Metadata metadata, in RealmValue value
                             throw new RealmException("Can't link to an embedded object that is already managed.");
                         }
 
+                        if (GetProperty(propertyName, metadata).Type.IsRealmValue())
+                        {
+                            throw new NotSupportedException("Embedded objects cannot be used as a RealmValue.");
+                        }
+
                         var embeddedHandle = CreateEmbeddedObjectForProperty(propertyName, metadata);
                         realm.ManageEmbedded(embeddedObj, embeddedHandle);
                         return;
+
+                    // Asymmetric objects will not reach this path unless they are used as a RealmValue.
+                    case IAsymmetricObject:
+                        throw new NotSupportedException($"Asymmetric objects cannot be linked to. Attempted to set {value} to {metadata.Schema.Name}.{propertyName}");
```

Comment:
> Attempted to set {value} to {metadata.Schema.Name}.{propertyName}

If you want to add this, you need to be consistent and have it in all the other exceptions.